include/swift/Demangling/Demangle.h

```diff
@@ -243,7 +243,8 @@ class Node {
     return getChild(getNumChildren() - 1);
   }
   NodePointer getChild(size_t index) const {
-    assert(getNumChildren() > index);
+    if (index >= getNumChildren())
+      return nullptr;
```

Comment:
This makes me slightly nervous if only because it'll potentially spread crashes from OOR accesses around more than the `assert()` does.

Reply:
Asserts get compiled out in release builds, so we'd be returning garbage instead. Improving this would be a good followup, though. Do we have a good way to distinguish between the use cases where we want failures to be tolerated, and the cases where we want failures to blow up immediately and obviously?

Reply:
I think calling `getChild()` with an OOR index was always an error previously. I take your point about asserts being compiled out in release builds, mind. Let's leave it returning `nullptr` for now — debugging isn't that difficult, after all :-)

Reply:
You realize the universe is going to curse us for that statement of yours....